# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/OffscreenCanvas.cpp
# Commit: 19fa8d4af66f
# Full Hash: 19fa8d4af66fb0dc9ace1dcf9cd365513573f05b
# Author: Jamie Nicol <jnicol@mozilla.com>
# Date: 2024-07-23 09:34:47
# Regressor Bug: 1898238
# File Overlap Count: 1
# Description:
#   Bug 1898238 - Don't call TexTypeForWebgl when initializing offscreen canvas. r=aosmond,jgilbert
#   
#   This is only used if the OffscreenCanvas has a webgl context, in which
#   case it forwards the tex type to WebGLContext::PresentFrontBuffer(). A
#   later patch in this series will make TexTypeForWebgl() return a
# ==============================================================================

diff -r 216ace67b0e2 -r 19fa8d4af66f dom/canvas/OffscreenCanvas.cpp
--- a/dom/canvas/OffscreenCanvas.cpp	Mon Jul 22 16:00:16 2024 +0000
+++ b/dom/canvas/OffscreenCanvas.cpp	Mon Jul 22 16:02:33 2024 +0000
@@ -34,13 +34,12 @@
 
 OffscreenCanvasCloneData::OffscreenCanvasCloneData(
     OffscreenCanvasDisplayHelper* aDisplay, uint32_t aWidth, uint32_t aHeight,
-    layers::LayersBackend aCompositorBackend, layers::TextureType aTextureType,
-    bool aNeutered, bool aIsWriteOnly, nsIPrincipal* aExpandedReader)
+    layers::LayersBackend aCompositorBackend, bool aNeutered, bool aIsWriteOnly,
+    nsIPrincipal* aExpandedReader)
     : mDisplay(aDisplay),
       mWidth(aWidth),
       mHeight(aHeight),
       mCompositorBackendType(aCompositorBackend),
-      mTextureType(aTextureType),
       mNeutered(aNeutered),
       mIsWriteOnly(aIsWriteOnly),
       mExpandedReader(aExpandedReader) {}
@@ -56,13 +55,12 @@
 
 OffscreenCanvas::OffscreenCanvas(
     nsIGlobalObject* aGlobal, uint32_t aWidth, uint32_t aHeight,
-    layers::LayersBackend aCompositorBackend, layers::TextureType aTextureType,
+    layers::LayersBackend aCompositorBackend,
     already_AddRefed<OffscreenCanvasDisplayHelper> aDisplay)
     : DOMEventTargetHelper(aGlobal),
       mWidth(aWidth),
       mHeight(aHeight),
       mCompositorBackendType(aCompositorBackend),
-      mTextureType(aTextureType),
       mDisplay(aDisplay) {}
 
 OffscreenCanvas::~OffscreenCanvas() {
@@ -321,7 +319,7 @@
   MOZ_ASSERT(mPendingCommit);
   mPendingCommit = nullptr;
   Maybe<OffscreenCanvasDisplayData> update = std::move(mPendingUpdate);
-  mDisplay->CommitFrameToCompositor(mCurrentContext, mTextureType, update);
+  mDisplay->CommitFrameToCompositor(mCurrentContext, update);
 }
 
 void OffscreenCanvas::CommitFrameToCompositor() {
@@ -338,7 +336,7 @@
   }
 
   Maybe<OffscreenCanvasDisplayData> update = std::move(mPendingUpdate);
-  mDisplay->CommitFrameToCompositor(mCurrentContext, mTextureType, update);
+  mDisplay->CommitFrameToCompositor(mCurrentContext, update);
 }
 
 UniquePtr<OffscreenCanvasCloneData> OffscreenCanvas::ToCloneData(
@@ -374,8 +372,8 @@
   }
 
   auto cloneData = MakeUnique<OffscreenCanvasCloneData>(
-      mDisplay, mWidth, mHeight, mCompositorBackendType, mTextureType,
-      mNeutered, mIsWriteOnly, mExpandedReader);
+      mDisplay, mWidth, mHeight, mCompositorBackendType, mNeutered,
+      mIsWriteOnly, mExpandedReader);
   SetNeutered();
   return cloneData;
 }
@@ -598,7 +596,7 @@
   MOZ_ASSERT(aData);
   RefPtr<OffscreenCanvas> wc = new OffscreenCanvas(
       aGlobal, aData->mWidth, aData->mHeight, aData->mCompositorBackendType,
-      aData->mTextureType, aData->mDisplay.forget());
+      aData->mDisplay.forget());
   if (aData->mNeutered) {
     wc->SetNeutered();
   }